pkg/trisa/keys/certs.go

```diff
@@ -0,0 +1,178 @@
+func FromProvider(certs *trust.Provider) (_ Key, err error) {
+	key := &Certificate{
+		privateKey: certs.GetKey(),
+	}
+
+	if key.certs, err = certs.GetLeafCertificate(); err != nil {
+		return nil, err
+	}
+	return key, nil
+}
+
+// Certificate is a wrapper for an x509 certificate (containing the public key) and
+// optionally for the private key associated with the certificate. This data structure
+// is used to create exchange keys using TRISA issued sealing-certificates. The public
+// key in the certificate will be used to send keys for sealing and the private key for
+// unsealing certificates.
+type Certificate struct {
+	certs      *x509.Certificate
+	privateKey interface{}
+}
+
+// Ensure the Certificate implements the Key interface.
+var _ Key = &Certificate{}
+
+// IsPrivate returns ture if there is a private key associated with the certificate.
+func (c *Certificate) IsPrivate() bool {
+	return c.privateKey != nil
+}
+
+// SealingKey is public key used to seal envelopes, usually an *rsa.PublicKey.
+func (c *Certificate) SealingKey() (interface{}, error) {
+	return c.certs.PublicKey, nil
+}
+
+// Proto returns the protocol buffer message with the public sealing key to exchange
+// with a remote counterparty to begin a TRISA transfer.
+func (c *Certificate) Proto() (msg *api.SigningKey, err error) {
+	msg = &api.SigningKey{
+		Version:            int64(c.certs.Version),
+		Signature:          c.certs.Signature,
+		SignatureAlgorithm: c.certs.SignatureAlgorithm.String(),
+		PublicKeyAlgorithm: c.PublicKeyAlgorithm(),
+		NotBefore:          c.certs.NotBefore.Format(time.RFC3339),
+		NotAfter:           c.certs.NotAfter.Format(time.RFC3339),
+	}
+
+	// TODO: should we marshal into PEM encoded form?
+	if msg.Data, err = x509.MarshalPKIXPublicKey(c.certs.PublicKey); err != nil {
+		return nil, fmt.Errorf("could not marshal pkix public key: %s", err)
+	}
+
+	return msg, nil
+}
+
+// UnsealingKey is the private key used to unseal envelopes, usually an *rsa.PrivateKey.
+// If no private key is available (IsPrivate() is false) then an error is returned.
+func (c *Certificate) UnsealingKey() (interface{}, error) {
+	if c.IsPrivate() {
+		return c.privateKey, nil
+	}
+	return nil, ErrNoPrivateKey
+}
+
+// PublicKeyAlgorithm refers to the public key algorithm of the x509.Certificate and
+// may be used to determine which cipher to use. Typically is "RSA".
+func (c *Certificate) PublicKeyAlgorithm() string {
+	return c.certs.PublicKeyAlgorithm.String()
+}
+
+// PublicKeySignature returns a unique identifier that can be used to manage public keys
+// and associate them with their counterpart private keys for unsealing.
+func (c *Certificate) PublicKeySignature() (string, error) {
+	return signature.New(c.certs.PublicKey)
+}
+
+// Marshal returns a PEM encoded Certificate Block along with a PEM encoded private key
+// block if one is available. This data is useful for storing Keys on disk or in a
```

Comment:
maybe "for storing *one's own* Keys on disk" ?

Reply:
Technically could also be used for storing counterparty public keys - but that's an edge use case. Updated to:

> This data is useful for storing Keys (particularly one's own private key pairs and sealing certificates) on disk or in a database, but should not be used to transfer or send keys.

Hopefully that helps clarify things a bit.